Installer/update-vs.ps1

```diff
@@ -0,0 +1,99 @@
+$installer_name = 'git-credential-manager.exe'
+$title = 'Git Credential Manager for Windows upater for Visual Studio v1.0'
+
+function check_access {
+  # Ensure that we're running elevated, and if we're not then restart as elevated.
+  $current_identity = [System.Security.Principal.WindowsIdentity]::GetCurrent()
+  $current_principle = new-object System.Security.Principal.WindowsPrincipal($current_identity)
+  $administrator_role = [System.Security.Principal.WindowsBuiltInRole]::Administrator
+  
+  if (-not ($current_principle.IsInRole($administrator_role))) {
+    write-host -ForegroundColor Red 'fatal: Access Denied'
+    write-host -ForegroundColor Yellow 'Administrator privilages are required, please run again with administrator privilages.'
+    
+    exit -2
+  }
+}
+
+$current_directory = get-location
+$script_name = $MyInvocation.MyCommand.Name
+$installer_path = "$current_directory\$installer_name"
+$script_path = "$current_directory\$script_name"
+$vs_installer_path = "${Env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer"
+
+write-host -ForegroundColor Cyan $title
+
+if (-not (test-path($installer_path))) {
+  write-host "fatal: unable to find $installer_name."
+  
+  write-host -NoNewLine 'Press any key to continue...';
+  $null = $Host.UI.RawUI.ReadKey('NoEcho,IncludeKeyDown');
+  
+  exit -1
+}
+
+if (-not (test-path($vs_installer_path))) {
+  write-host -ForegroundColor Yellow 'Visual Studio was not detected on this machine.'
+  
+  write-host -NoNewLine 'Press any key to continue...';
+  $null = $Host.UI.RawUI.ReadKey('NoEcho,IncludeKeyDown');
+  
+  exit 1
+}
+
+# Ensure that this script is executing elevated
+check_access
+
+write-host 'Scanning for instllations of Visual Studio to update'
+write-host 
+
+$vswhere_path = "$vs_installer_path\vswhere.exe"
+$vswhere_cmd = "& '$vswhere_path' -all -prerelease -nologo -utf8 -version '[15.0, 16.0)'"
+
+# Run vswhere and parse its output.
```

Comment:
Did you consider the more idiomatic:

```ps1
$installations = (& $vswhere_path -all -prerelease -nologo -format json) | convertfrom-json | foreach-object {$_} | select installationname, installationpath
```
That would simplify the upcoming loop by letting us iterate fully-formed objects instead parsing line-by-line.

Reply:
um... I have no idea how. So line-by-line is how I did it. 🤷🏻‍♂️

How would one get fully formed objects out of vswhere?